scripts/sysman/firmware.yml

```diff
@@ -90,6 +90,49 @@ returns:
     - $X_RESULT_ERROR_INSUFFICIENT_PERMISSIONS:
         - "User does not have permissions to perform this operation."
 --- #--------------------------------------------------------------------------
+type: function
+desc: "Find Firmware Flash Progress"
+version: "1.7"
+class: $sFirmware
+name: UpdateProgress
+details:
+    - "The application may call this function from simultaneous threads."
+    - "The implementation of this function should be lock-free."
+params:
+    - type: $s_firmware_handle_t
+      name: hFirmware
+      desc: "[in] Handle for the component."
+    - type: "uint32_t*"
+      name: pCompletionPercent
+      desc: "[in,out] Pointer to the Completion Percentage of Firmware Update"
+returns:
+    - $X_RESULT_ERROR_INSUFFICIENT_PERMISSIONS:
+        - "User does not have permissions to perform this operation"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Collect Firmware Logs"
+version: "1.7"
+class: $sFirmware
+name: Logs
+details:
+    - "The caller can pass nullptr for pFirmwareLog when querying only for size."
+    - "The caller must provide memory for Firmware log."
+    - "The application may call this function from simultaneous threads."
+    - "The implementation of this function should be lock-free."
+params:
+    - type: $s_firmware_handle_t
+      name: hFirmware
+      desc: "[in] Handle for the component."
+    - type: "size_t*"
+      name: pSize
+      desc: "[in,out] size of firmware log"
```

Comment:
please specify what happens if user passes a size that is lesser or greater than available.

Reply:
Logs will be filled only for the allocated buffer.

Reply:
up to you. Whether we can return the whole log or just partial for size less than available. But we need to specify what happens. 

this

> Logs will be filled only for the allocated buffer.

is what we usually do in other APIs.